### PR TITLE
Switch to PCL library and more

### PR DESCRIPTION
**Pull request by lploumen**
- Switched to PCL library.
- Using an xml configuration file for the example project, this way credentials aren't included in the project.
- Using AngleSharp instead of HtmlAgilityPack since I found the css selector syntax more clear than the linq syntax of HtmlAgilitypath (Xpath is not available in PCL HtmlAgilityPack).
- Using async method inside the library but at the moment they are used synchronously, next step is to use async method in the interface. (I didn't want to change this now).
- Parsing updated for geocaching.com. A div id has changed.
- Also updated some webpages, but with my username... We need to find a way to be consistent with this...

This is the first time I use Git and a pull request so any comments are welcome ;)

### CODE REVIEW

**Comment by krzema12** [comment]
First of all, it's super-awesome that you put so much effort into this! Really appreciate it.

I skimmed through your commits.

The first thing that is clearly visible that you changed all tabs into spaces. I like the idea in general (from some time, I prefer spaces, too), but it shouldn't be changed along with some real code changes because it makes the changes hard to review. I cannot see what you really changed.

Second thing, let's pay attention to make the commits self-contained. What I mean is that you shouldn't e.g. create "ConsoleApplication1" in the first commit and delete it in the next one. You can just squash these two commits together and then you won't need the second commit. Another example: you're adding some missing nuget packages in a separate commit. The previous commit should contain this change, right? Without it, it shouldn't compile. Remember that you can actually edit your commit.

Your commits should also change one thing, aspect. So for example your first commit (about XML config), looking at its title - it just makes sense, you move the config to XML and add necessary mechanisms to read data from XML. But "First step toward a PCL library" actually contains several changes. You could move to AngleSharp in a separate commit. There are also some changes (not related to switching to PCL?) in OCConnector and WebBrowserSimulator. I also don't get why you commented out "GetTrackableDetails".

You're new to git - it's totally fine not to prepare perfect commits, it's just something you constantly learn (my commits are not perfect, too). A good training is to look at each of your commit (diff -> what is changed) and see if it's readable, if the change really makes sense. 10 small commits are better than 1 big because it's easier to see what's actually changed.

To recap: I've just changed all tabs to spaces. Please rebase your commits onto my change and divide them appropriately so that they're easier to review. Then we can discuss about them further.

**Comment by lploumen** [comment]
Hi,
OK I'll see what I can do. Thank you for your remarks.
Problem is that the PCL switch involves many things to make to project compile (changes in WebbrowserSimulator, OC tokens, AngleSharp,...). They must be done in one step to keep the project compiling (For example, HtmAgilityPack won't work with XPath syntax).
Is it a problem to have a commit with all the changes related to the PCL switch ? I mean I can split them but the code won't compile until all the PCL switch is done...
I clearly understand your other remarks but now I have to dig into the git documentation ;)

**Comment by lploumen** [comment]
Could you give me some guidance regarding reordering and reviewing my commits ? I must admit I'm a bit lost at the moment.
Based on the current state of my repository, is it better to clean the repository, check commits and sync ? (=Start over) or reorder everything using git commands ? If so could you give me some hints ?

**Comment by krzema12** [comment]
For sure don't start over completely, you'll lose quite much work.
First of all, it would be awesome if you learned some basics of git. You should know how to edit your commits ("git amend"), what a rebase and a cherry-pick is. It's really hard for me to explain all necessary steps you should do in git. Here's a nice chapter from "Pro Git" book: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History . The book in general is good. If you're totally new to git, I'd simply start from Chapter 1.

I'm sorry, I'm quite passive now regarding the project, there's a lot going on in my private life and to find some time to look here on GitHub - trust me, it's quite challenging.

**Comment by krzema12** [comment]
About dividing the commits - I think you can split the big commit into smaller pieces. Let's take a look at switching from HtmlAgilityPack to AngularSharp. I suppose that AngularSharp works also with non-PCL library, right? So you can treat changing this framework as a separate change -> separate commit. The ultimate aim is to convert the library to PCL, but who said that switching to PCL must be done in one commit? You can do some preparations in separate commits, I think :)

I hope you get my point, I'm not sure if my explanations are clear.

Anyway, it would be much easier to talk about your commits if there weren't this tabs/spaces issue. So maybe the commit is in fact not so big, only it appears so.

**Comment by lploumen** [comment]
Hi Peter,
Regarding the tab/space issue I'll check was is going on. I didn't set anything anywhere :) Maybe is it Visual studio or Resharper doing something here. I'll have a look at it.
